File: repository/src/main/java/com/SReMake/repository/system/impl/RoleResourcesRepositoryImpl.java
package com.SReMake.repository.system.impl;

//import com.SReMake.model.Tables;
//import com.SReMake.model.system.Resources;
//import com.SReMake.model.system.RoleResources;
//import com.SReMake.repository.system.RoleResourcesRepository;
//import org.babyfish.jimmer.spring.repo.support.AbstractJavaRepository;
//import org.babyfish.jimmer.sql.JSqlClient;
//import org.springframework.stereotype.Repository;
//
//import java.util.Collection;
//import java.util.List;
//
//@Repository
//public class RoleResourcesRepositoryImpl extends AbstractJavaRepository<RoleResources, Long> implements RoleResourcesRepository {
//    public RoleResourcesRepositoryImpl(JSqlClient sql) {
//        super(sql);
//    }
//
//
//    @Override
//    public List<Resources> listByRole(Collection<Long> roleIds) {
//
//        return this.sql.createQuery(Tables.ROLE_RESOURCES_TABLE)
//                .where(Tables.ROLE_RESOURCES_TABLE.roleId().in(roleIds))
//                .select(Tables.ROLE_RESOURCES_TABLE.resources())
//                .fetchOptional().stream().toList();
//    }
//
//    @Override
//    public void deleteByRoleIdAndResourceId(Long roleId, Collection<Long> resourceId) {
//        this.sql.createDelete(Tables.ROLE_RESOURCES_TABLE).where(Tables.ROLE_RESOURCES_TABLE.roleId().eq(roleId), Tables.ROLE_RESOURCES_TABLE.resourcesId().in(resourceId)).execute();
//    }
//}

File: repository/src/main/kotlin/com/SReMake/repository/system/impl/RoleResourcesRepositoryImpl.kt
package com.SReMake.repository.system.impl

import com.SReMake.model.system.RoleResources
import com.SReMake.model.system.resource
import com.SReMake.model.system.resourceId
import com.SReMake.model.system.roleId
import com.SReMake.repository.system.RoleResourcesRepository
import org.babyfish.jimmer.spring.repo.support.AbstractKotlinRepository
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.ast.expression.eq
import org.babyfish.jimmer.sql.kt.ast.expression.valueIn
import org.springframework.stereotype.Repository

@Repository
class RoleResourcesRepositoryImpl(sql: KSqlClient) : RoleResourcesRepository,
    AbstractKotlinRepository<RoleResources, Long>(sql) {
    override fun listByRole(roleIds: Collection<Long>) =
        this.sql.createQuery(RoleResources::class) {
            where(table.roleId.valueIn(roleIds))
            select(table.resource)
        }.execute()

    override fun deleteByRoleIdAndResourceId(roleId: Long, resourceIds: Collection<Long>) {
        this.sql.createDelete(RoleResources::class) {
            where(table.roleId.eq(roleId), table.resourceId.valueIn(resourceIds))
        }.execute()
    }

}